components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -1786,12 +1468,556 @@ subroutine ma_convproc_tend(                                           &
+   integer, intent(in)  :: ktop                 ! top level index
+   integer, intent(in)  :: kbot                 ! bottom level index
+   integer, intent(in)  :: iconvtype            ! 1=deep, 2=uw shallow
+   real(r8),intent(in)  :: dt                   ! Model timestep [s]
+   real(r8),intent(in)  :: mbsth                ! threshold below which we treat the mass fluxes as zero [mb/s]
+   real(r8),intent(in)  :: dp_i(pver)           ! dp [mb]
+   real(r8),intent(in)  :: dpdry_i(pver)        ! dp [mb]
+   real(r8),intent(in)  :: cldfrac(pcols, pver) ! cloud fraction [fraction]
+   real(r8),intent(in)  :: rhoair_i(pver)       ! air density at current i [kg/m3]
+   real(r8),intent(in)  :: zmagl(pver)          ! height above surface [m]
+   real(r8),intent(in)  :: dz                   ! layer thickness [m]
+   real(r8),intent(in)  :: mu_i(pverp)          ! mu at current i (note pverp dimension, see ma_convproc_tend) [mb/s]
+   real(r8),intent(in)  :: eudp(pver)           ! eu(i,k)*dp(i,k) at current i [mb/s]
+   real(r8),intent(in)  :: const(pcnst_extd,pver)   ! gathered tracer array [kg/kg]
+
+   real(r8), intent(in) :: t(pcols,pver)        ! Temperature [K]
+   real(r8), intent(in) :: aqfrac(pcnst_extd)   ! aqueous fraction of constituent in updraft [fraction]
+   real(r8), intent(in) :: icwmr(pcols,pver)    ! Convective cloud water from zm scheme [kg/kg]
+   real(r8), intent(in) :: rprd(pcols,pver)     ! Convective precipitation formation rate [kg/kg/s]
+   real(r8), intent(out) :: dconudt_wetdep(pcnst_extd,pverp) ! d(conu)/dt by wet removal[kg/kg/s]
+   real(r8), intent(out) :: dconudt_activa(pcnst_extd,pverp) ! d(conu)/dt by activation [kg/kg/s]
+   real(r8), intent(out) :: fa_u(pver)           ! fractional area of in the updraft
+   real(r8), intent(inout) :: conu(pcnst_extd,pverp)   ! mix ratio in updraft at interfaces [kg/kg]
+
+   ! local variables
+   integer      :: icnst        ! index of pcnst_extd
+   integer      :: kk           ! index of vertical level
+   integer      :: kp1          ! index of kk+1
+   integer      :: kactcnt      ! Counter for no. of levels having activation
+   integer      :: kactfirst    ! Lowest layer with activation (= cloudbase)
+   real(r8)     :: clw_cut              ! threshold clw value for doing updraft [kg/kg]
+   real(r8)     :: dt_u(pver)           ! lagrangian transport time in the updraft [s]
+   real(r8)     :: mu_p_eudp(pver)      ! = mu_i(kp1) + eudp(k) [mb/s]
+   real(r8)     :: cldfrac_i(pver)      ! cldfrac at current icol (with adjustments) [fraction]
+   real(r8)     :: wup(pver)            ! mean updraft vertical velocity at current level updraft [m/s]
+   real(r8)     :: f_ent                ! fraction of updraft massflux that was entrained across this layer == eudp/mu_p_eudp [fraction]
+
+    ! initiate variables
+    kactcnt = 0 ; kactfirst = 1
+    wup(:) = 0.0
+    clw_cut = 1.0e-6    ! Skip levels where icwmr(icol,k) <= clw_cut (=1.0e-6) to
```

Comment:
Declare this as parameter, since it seems to be a constant

Reply:
will do in this PR